### ISSUE TITLE
False positive in `indentation_linter()`?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1800]
Possibly related to https://github.com/r-lib/styler/issues/1065.

``` r
library(lintr)

'test_that("bla", {
  x <-
    letters[1:3] %>%
    paste0() %>%
    length()
})' -> code

lint(
  text = code,
  linters = indentation_linter()
)
#> <text>:4:4: style: [indentation_linter] Indentation should be 6 spaces but is 4 spaces.
#>     paste0() %>%
#>    ^~~
```

<sup>Created on 2022-12-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

This is what `{lintr}` expects:

```r
test_that("bla", {
  x <-
    letters[1:3] %>%
      paste0() %>%
      length()
})
```

But `{styler}` changes this to the above-mentioned version that produces lints.

**Comment by IndrajeetPatil** [comment]
Thanks for the explanation. 

Makes sense to not have a trailing `<-`, but note that [the style guide](https://style.tidyverse.org/pipes.html#assignment-2) specifically mentions this as an acceptable way of doing assignments in piped workflows:

<img width="733" alt="Screenshot 2022-12-05 at 19 40 39" src="https://user-images.githubusercontent.com/11330453/205717207-5d991d31-afbb-4a65-81fd-8b16fdc20592.png">

Why is this important?

The `identation_linter()` is part of the default set of linters, which means that anyone who is using this style of assignment will see lints from this linter. Add this to the fact that the new GHA workflow for linting produces errors if any lints are found.

This means that users are going to see build failures for writing tidyverse code in a style that the style guide explicitly says is acceptable. IMO, this is surprising and frustrating for the user.

We have a few options out of this quagmire: 

- support this style of assignment (i.e., no lint from `identation_linter()`)
- provide a parameter to `identation_linter()` to toggle on and off whether this assignment style should be linted
- ask the style guide to be revised

**Comment by AshesITR** [comment]
I think an optional argument (defaulting to tidyverse style) is in order.
The refactoring necessary for this isn't trivial, though :(

**Comment by MichaelChirico** [comment]
Q is, is this worth demoting `indentation_linter()` out of defaults for the next release? e.g. marking it as "WIP" and "slated-for-default", or do we think it's already consistent enough with the style guide to be released as a default with some known edge case like this?
